test/parser_test.exs

```diff
@@ -0,0 +1,236 @@
+defmodule ParserTest do
+  use ExUnit.Case, async: true
+
+  import Membrane.Ogg.Parser
+  alias Membrane.Ogg.Parser.Packet
+
+  defp create_page(segments) do
+    segment_table = Enum.map(segments, fn seg -> <<seg>> end) |> Enum.join()
+    content = Enum.map(segments, fn seg -> segment(seg) end) |> Enum.join()
+    before_crc = <<"OggS", 0, 0, 0::size(8 * 8), 0::size(4 * 8), 0::size(4 * 8)>>
+    after_crc = <<Enum.count(segments)>> <> segment_table <> content
+    crc_payload = before_crc <> <<0::size(32)>> <> after_crc
+
+    crc =
+      CRC.crc(
+        %{
+          extend: :crc_32,
+          init: 0x0,
+          poly: 0x04C11DB7,
+          xorout: 0x0,
+          refin: false,
+          refout: false
+        },
+        crc_payload
+      )
+
+    before_crc <> <<crc::little-unsigned-size(32)>> <> after_crc
+  end
+
+  defp create_page_with_invalid_crc(segments) do
+    <<before_crc::binary-size(22), _crc::binary-size(4), after_crc::binary>> =
+      create_page(segments)
+
+    before_crc <> <<0::size(32)>> <> after_crc
+  end
+
+  defp create_page_with_invalid_header(segments) do
+    <<"OggS", rest::binary>> = create_page(segments)
+    <<"RanD">> <> rest
+  end
+
+  defp segment(len) do
+    <<len::size(len * 8)>>
+  end
+
+  test "Parses a simple page with a single segment" do
```

Comment:
We had internal discussion recently where we decided to favor shorter test names so e.g.

```suggestion
    test "simple page with a single segment
```

or 

```elixir
    test "simple page with multiple segments"
    test "too short input"
    test "multiple pages"
    test "page with multiple packets"
    # etc.
```